L1Trigger/L1THGCal/src/backend/HGCalMulticlusteringHistoImpl.cc

```diff
@@ -5,6 +5,7 @@
 
 HGCalMulticlusteringHistoImpl::HGCalMulticlusteringHistoImpl( const edm::ParameterSet& conf ) :
     dr_(conf.getParameter<double>("dR_multicluster")),
+    dr_byLayer_(conf.existsAs<std::vector<double>>("dR_multicluster_byLayer") ? conf.getParameter<std::vector<double>>("dR_multicluster_byLayer") : std::vector<double>()),
```

Comment:
It would be better to check the size of the vector. Using e.g.  `HGCalTriggerTools::lastLayerBH()`.

Reply:
I can't check at creation time because the triggerTools only knows the layers after it has been initialized with the eventSetup.
I can put the check whe the event setup is passed

Reply:
Ah, right. Yes, it's probably cleaner to have the check in the eventSetup() rather than waiting an exception when `dr_byLayer_.at()` is called.